docs/tests/test_generated_docs.py

```diff
@@ -0,0 +1,20 @@
+"""Tests for the generated documentation."""
+
+import pathlib
+import subprocess
+import unittest
+
+import git
+
+
+class GeneratedDocumentationTest(unittest.TestCase):
+    """Tests for the generated documentation."""
+
+    def test_up_to_dateness(self):
+        """Tests that the workspace is clean after generting the docs."""
+        docs = pathlib.Path(__file__).resolve().parent.parent
+        repo = git.Repo(docs.parent)
+        create_metrics_and_sources = docs / "src" / "create_metrics_and_sources_md.py"
+        subprocess.run(["python3", create_metrics_and_sources], check=True)  # skipcq: BAN-B603,BAN-B607
+        metrics_and_sources = docs / "METRICS_AND_SOURCES.md"
```

Comment:
And automatically commit the changed docs? I think I would prefer a pre-commit hook so that the updated docs are part of the same commit as the changes that caused it.

Reply:
I've added updating the generated documentation to the pre-commit hook. You'll need to copy git-hooks/pre-commit to .git/hooks to effectuate it.